### PR TITLE
feat: add `round` function

### PR DESCRIPTION
**Pull request by shan-shaji**
> [!TIP]
> The owner of this PR can publish a _preview release_ by commenting `/publish` in this PR. Afterwards, anyone can try it out by running `pnpm add radashi@pr<PR_NUMBER>`.

## Summary
The round function is designed to round a given number to a specified precision. 

## Related issue, if any:

https://github.com/sodiray/radash/issues/61

## What kind of change does this PR introduce?
Feature

## For any code change,

- [x] Related documentation has been updated, if needed
- [x] Related tests have been added or updated, if needed

## Does this PR introduce a breaking change?

No

### CODE REVIEW

File: src/number/tests/round.test.ts
Comment (localusercamp) [inline src/number/tests/round.test.ts:1]: I think we need more tests for edge-cases like 
`round(1.01, 1000)`
`round(1.01, -1000)`

**Review by localusercamp — CHANGES_REQUESTED** [review]

File: src/number/round.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ *
+ * round(4.006)
+ * // => 4.01
+ *
+ * round(1234.56, -2);
+ * // => 1200
+ *
+ *
+ */
+export function round(value: number, precision = 2): number {
```
[the remark this answers is not in the document]
Comment (shan-shaji) [reply src/number/round.ts:17]: Cool i will update it. Thanks @localusercamp

File: src/number/round.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ *
+ */
+export function round(value: number, precision?: number): number {
+  precision =
+    precision == null
+      ? 0
+      : precision >= 0
+        ? Math.min(precision, 307)
+        : Math.max(precision, -323)
```
Comment (aleclarson) [inline src/number/round.ts:23]: How can we verify these precision limits to be correct?

Interestingly, Lodash does this:
```ts
precision = precision == null ? 0 : (precision >= 0 ? Math.min(precision, 292) : Math.max(precision, -292))
```

Comment (shan-shaji) [reply src/number/round.ts:23]: @aleclarson  I added that check inspired from them. 
Screenshot 2024-06-30 at 10 34 34 PM

Comment (shan-shaji) [reply src/number/round.ts:23]: Also i found an issue related with this: https://github.com/lodash/lodash/issues/4232

Comment (aleclarson) [reply src/number/round.ts:23]: Here's what Claude Sonnet 3.5 says on the matter:

> You're right to question this. Lodash's choice of 292 is indeed intentional and has a specific reason. Let me explain:
> 
> 1. Precision: While JavaScript can theoretically represent numbers up to 308 decimal places, it doesn't maintain full precision for all of those digits.
> 
> 2. Reliable accuracy: Lodash uses 292 as a more conservative limit to ensure reliable and consistent results across different JavaScript engines and environments.
> 
> 3. IEEE 754 standard: This limit is related to the IEEE 754 double-precision floating-point format, which guarantees about 15-17 decimal digits of precision.
> 
> 4. Safety margin: 292 provides a safety margin, avoiding potential rounding errors or inconsistencies that might occur when pushing to the absolute limits of JavaScript's number representation.
> 
> So, while 308 is technically the maximum exponent, Lodash's choice of 292 is a pragmatic one, prioritizing consistent and reliable results over the theoretical maximum. For most practical applications, this limit is more than sufficient and helps prevent subtle bugs that could arise from pushing the boundaries of floating-point precision.

I will push an update, along with the Lodash test suite and a reimplementation to support the edge cases that `lodash.round` does.

Comment (shan-shaji) [reply src/number/round.ts:23]: Thanks @aleclarson  for the explanation. So they chose 292 to ensure results are reliable and consistent across different JavaScript engines (e.g., V8 in Chrome, SpiderMonkey in Firefox) and environments (e.g., browsers, Node.js).

Comment (aleclarson) [reply src/number/round.ts:23]: I tracked down the original commit:
68ae6fc2122c4ea23ae42f9b176e20cffde2d7ba

Not much justification is given. If we can use technically “more correct” limits, we should as long as `NaN` is avoided. If someone reports misbehavior in some JS engine, we'll consider a breaking change then, but considering such large precision values are probably very rare in Javascript code, that won't happen anytime soon.

Comment (shan-shaji) [reply src/number/round.ts:23]: So to be on the safe side shall we stick with 292 then?

Comment (aleclarson) [reply src/number/round.ts]: After some testing, I found that `292` is the upper limit that prevents NaN in all cases, while `-323` works fine as a lower limit in all cases.

**Comment by aleclarson** [comment]
Thanks for the contribution! Hope to see more from you in the future.
